malaria-app-android/src/main/java/com/peacecorps/malaria/activities/RapidFireGame.java

```diff
@@ -236,17 +241,31 @@ public RapidFireTimeCounter(long millisInFuture, long countDownInterval) {
         @Override
         public void onTick(long l) {
             timer.setText(""+l/1000);
+            millisLeft=l;
         }
 
         @Override
         public void onFinish() {
             counter.cancel();
             prepNextQues();
         }
+    }
 
-
+    @Override
+    protected void onPause() {
+        counter.cancel();
+        counter= null;
+        super.onPause();
     }
 
+    @Override
+    protected void onResume() {
+        if(counter==null){
+            counter= new RapidFireTimeCounter(millisLeft,1000);
```

Comment:
I do not have a problem with the null check. The timer does not need to set to null and reinitialised is all I am saying. You can cancel it in onPause and then in onResume set the time from sharedPreffs and start it again.